File: 3932-count-k-th-roots-in-a-range/3932-count-k-th-roots-in-a-range.swift
class Solution {
    func countKthRoots(_ l: Int, _ r: Int, _ k: Int) -> Int {
        if k == 1 {
            return r - l + 1
        }

        var result = 0
        var x = 0

        while true {
            var value = 1

            for _ in 0..<k {
                value *= x
                if value > r {
                    return result
                }
            }

            if value >= l {
                result += 1
            }

            x += 1
        }
    }
}

File: 3932-count-k-th-roots-in-a-range/3932-count-k-th-roots-in-a-range.scala
object Solution {
    def countKthRoots(l: Int, r: Int, k: Int): Int = {
        if (k == 1) {
            return r - l + 1
        }

        var result = 0
        var x = 0

        while (true) {
            var value = 1L
            var i = 0

            while (i < k) {
                value *= x
                if (value > r) {
                    return result
                }
                i += 1
            }

            if (value >= l) {
                result += 1
            }

            x += 1
        }

        result
    }
}